### ISSUE TITLE
T - Acct Settings: user details component form

### ISSUE THREAD

**Issue by tobinmori** [mdn/kuma#7097]
# Acceptance criteria

- [ ] user form with name/static email/username/lang (slide 2)

# Q:
- [ ] check language dropdown size

**Comment by tobinmori** [comment]
Moving the conversation here out of the Gdoc:

@schalkneethling mentioned moving email completely out of this form now since we have a specific component for managing this. Is that ok @atopal?

**Comment by atopal** [comment]
No, I think the email part should stay in the account settings, email in the side bar should be for managing receiving MDN emails, and should probably called "emails" instead.

**Comment by tobinmori** [comment]
status: not started

**Comment by schalkneethling** [comment]
Here is the latest design:

![Screenshot 2020-06-03 at 10 30 08](https://user-images.githubusercontent.com/10350960/83614346-5c89f580-a585-11ea-84b0-be5d14864f25.png)

**Comment by schalkneethling** [comment]
@peterbe @escattone For the above form we need the following fields:

## Fields

```
avatarUrl
email
isNewsletterSubscriber
username
fullname
language
```

### What whoami currently has:

```
avatarUrl
email
username
```

Should the missing fields be added to `whoami` or should there be a separate API endpoint like `get_user_details`?

## Languages(locales)

We also need the list of all the locales(languages) we support. Here again, should we have an API endpoint like, `get_supported_locales`?

## Updating user details

When posting the form, do we have an existing endpoint to use?

**Comment by peterbe** [comment]
> Should the missing fields be added to `whoami` or should there be a separate API endpoint like `get_user_details`?

Definitely a new endpoint. The `whoami` always has to be as slim as humanly possible. Things like `fullname` is not needed on every page. But things like `avatar_url` is.

**Comment by peterbe** [comment]
Actually, "slim as humanly possible" was maybe a bit exaggerated. A minority of people are signed in and we're currently including things like `subscriber_number` and `is_subscriber` in the `whoami` which we're only ever using on the subscription landing page. Arguably we can refactor that away but I think it was put there because we might want to display the mugshot and the drop-down (that appears when you press your mugshot) depending on you being a subscriber or not. (E.g. I've always thought the mugshot picture should get a gold medal or a royal crown overlay if you're a subscriber)

Anyway, since we're going to have a dedicated endpoint for the PUT and, we might as well have a dedicated endpoint of the GET.

**Comment by peterbe** [comment]
> When posting the form, do we have an existing endpoint to use?

It'll probably be the same endpoint you use to fetch the data. E.g.

```javascript
const userDetails = await (await fetch('/api/v1/account/')).json()
...

await fetch('/api/v1/account/', {
  method: "PUT", 
  headers: {
                'X-CSRFToken': getCookie('csrftoken'),
            }
   })
```

Which reminds me, what did we decide the addressbar URL is going to be when you click "Account Settings"? It doesn't matter much but it might be nice that the v1 API endpoint somewhat resembles that. I.e. the topic of calling it "account" or "user" or "accountsettings" etc.

**Comment by peterbe** [comment]
> We also need the list of all the locales(languages) we support.

This should be done from jinja. Look at how `kuma/payments/jinja2/payments/index.html` works. 
You can follow the trace of the `next_subscriber_number` variable and see how it goes from the Django view into the React component. 

The day we port Account Settings to Yari we might reconsider this so that the choices are driven from Kuma rather than the SPA in Yari. But that's a much different day.

**Comment by schalkneethling** [comment]
Thank you for all of the feedback @peterbe
